### PR TITLE
Toasts-Loaders-404 Page

### PR DESCRIPTION
**Pull request by RakshaPawar108**
- Added Loader before fetching cart, products and wishlist
- Removed alerts from all pages and added relevant toasts
- Added 404 route and not found page

**Note: Deploy preview is currently not available since the `dev` branch is up for submission so I cannot make changes or merges to the` dev` branch and I am merging this branch into the `temp-dev` branch.**

**PS: I have hosted the `temp-dev` branch separately, so deploy previews will start coming from the next PR that will be merged into `temp-dev`**

### CODE REVIEW

File: src/pages/Cart/Cart.css [hunk 1]
```diff
@@ -1,6 +1,7 @@
 .cart-wrapper {
   width: 90%;
   margin: 5rem auto 7rem auto;
+  height: 100vh;
```
Comment (gadgetfather) [inline src/pages/Cart/Cart.css]: use `min-height` if this is a page wrapper. other wise content will overflow if someone adds too many things

File: src/pages/ProductListing/ProductListing.css [hunk 2]
```diff
@@ -35,21 +35,21 @@
   color: var(--dark-gray);
 }
 
-.prod-action-btn:disabled{
-  background-color:var(--dark-gray);
+.prod-action-btn:disabled {
+  background-color: var(--dark-gray);
```
Comment (gadgetfather) [inline src/pages/ProductListing/ProductListing.css:39]: if possible try to name color by element where you are using it. avoid using direct color names as variables. or even if you want to use a color name then specify the element before it. 
Attaching screenshot of Github's color variables (This is from the same page of code review)
image

File: src/pages/Wishlist/Wishlist.css [hunk 3]
```diff
@@ -5,6 +5,8 @@ body {
 .wishlist-wrapper {
   width: 90%;
   margin: 5rem auto 7rem auto;
+  height: 100vh;
```
Comment (gadgetfather) [inline src/pages/Wishlist/Wishlist.css]: as suggested above , if this is page wrapper then use min-height

**Review by gadgetfather — APPROVED** [review]
- Code is clean 
I have suggested some things. you can check the comments.
Also if the preview is not possible, you can share the gif if possible.
Apart from that Great work @RakshaPawar108

File: src/pages/Wishlist/Wishlist.css (hunk 3 above)
Comment (RakshaPawar108) [reply src/pages/Wishlist/Wishlist.css]: Done

File: src/pages/Cart/Cart.css (hunk 1 above)
Comment (RakshaPawar108) [reply src/pages/Cart/Cart.css]: Done

**Pull request merged**